File: CarSharing.FrontEnd/src/app/models/reservation.ts
import { Car } from "./car";
import { Client } from "./customer";
import { Fine } from "./fine";

export class Booking {
    public id!: number;
    public customer!: Client;
    public car!: Car;

    public startDate!: Date;
    public expectedReturnDate!: Date;
    public actualReturnDate?: Date;

    public totalSum!: number;

    public fines: Fine[] = [];
}

File: CarSharing.Domain/DTOs/Reservation.cs
namespace CarSharing.Domain.DTOs;

public class Reservation
{
    public int Id { get; set; }

    public DateTime StartDate { get; set; }

    public DateTime ExpectedReturnDate { get; set; }

    public DateTime? ActualReturnDate { get; set; }

    public Car Car { get; set; }

    public Customer Customer { get; set; }

    public List<Fine> Fines { get; set; }
    
    public double TotalSum { get; set; }
}